validations/src/validations.rs

```diff
@@ -150,30 +155,162 @@ pub fn validate_rad_request(rad_request: &RADRequest) -> Result<(), failure::Err
+                    RadonTypes::Float(_) => {
+                        update_liars(&mut liars, rad_types, counter.max_pos == 3)
+                    }
+                    RadonTypes::Integer(_) => {
+                        update_liars(&mut liars, rad_types, counter.max_pos == 4)
+                    }
+                    RadonTypes::Map(_) => update_liars(&mut liars, rad_types, counter.max_pos == 5),
+                    RadonTypes::String(_) => {
+                        update_liars(&mut liars, rad_types, counter.max_pos == 6)
+                    }
+                },
+            })
+            .collect();
+
+        Ok((results, liars))
+    } else {
+        let errors: Vec<RadError> = reveals
+            .into_iter()
+            .filter_map(|reveal| match reveal.into_inner() {
+                Err(e) => Some(e),
+                Ok(_) => None,
+            })
+            .collect();
+
+        // TODO: Currently we return the first error, but we should use the mode of errors
+        if !errors.is_empty() {
+            Err(errors[0].clone())
+        } else {
+            Err(RadError::default())
+        }
+    }
+}
+
 /// Function to validate a tally consensus
 pub fn validate_consensus(
     reveals: Vec<&RevealTransaction>,
     miner_tally: &[u8],
     consensus: &RADTally,
+    non_error_min: f64,
 ) -> Result<HashSet<PublicKeyHash>, failure::Error> {
-    let radon_types_vec: Vec<RadonTypes> = reveals
-        .iter()
-        .filter_map(|reveal| RadonTypes::try_from(reveal.body.reveal.as_slice()).ok())
-        .collect();
+    let results = serial_iter_decode(
+        &mut reveals
+            .iter()
+            .map(|&reveal_tx| (reveal_tx.body.reveal.as_slice(), reveal_tx)),
+        |e: RadError, slice: &[u8], reveal_tx: &RevealTransaction| {
+            log::warn!(
+                "Could not decode reveal from {:?} (revealed bytes were `{:?}`): {:?}",
+                reveal_tx,
+                &slice,
+                e
+            );
+            Some(RadonReport::from_result(Err(e), &ReportContext::default()).unwrap())
+        },
+    );
+
+    let len_results = results.len();
+    let clause_result = tally_precondition_clause(results, non_error_min);
+
+    let report = match clause_result {
+        Ok((radon_types_vec, liars)) => run_tally_report(radon_types_vec, consensus, Some(liars))?,
+        Err(e) => {
+            let context = &mut ReportContext::default();
+            let mut metadata = TallyMetaData::default();
+            metadata.liars = vec![false; len_results];
+            context.stage = Stage::Tally(metadata);
+
+            RadonReport::from_result(Err(e), &context)?
+        }
+    };
 
-    let report = run_tally_report(radon_types_vec, consensus)?;
     let metadata = report.metadata.clone();
-    let tally_consensus = report
-        .into_inner()
-        .and_then(|tally| RadonTypes::encode(&tally))?;
+    let tally_consensus: Vec<u8> = Vec::<u8>::try_from(&report)?;
```

Comment:
You can use some magic syntax to avoid specifying the type twice, but I think it's more clear to just use `try_into`:

```suggestion
    let tally_consensus: Vec<u8> = <_>::try_from(&report)?;
    let tally_consensus: Vec<u8> = (&report).try_into()?;
```